### PR TITLE
Implicit param

### PR DESCRIPTION
**Pull request by iTakeshi**
Currently this plugin fails with a function which has curried implicit parameters (ignoring them)

```scala
// src
@identity def add(a: Int)(implicit z: Int): Int = a + z
// is converted to
def add(a: Int): Int = a.+(z) // crash!! z is undefined
```

This is because `LogicalTree` filters out implicit parameters which does not have certain prefix (`nme.EVIDENCE_PARAM_PREFIX`), and I'm not sure why such filter is necessary.
Therefore this PR suggests to remove this filter, but if it is necessary, could you explain why?

CC @xeno-by and @Dveim

// copied from #50

### CODE REVIEW

File: plugin/src/main/scala/org/scalameta/paradise/reflect/LogicalTrees.scala [hunk 1]
```diff
@@ -898,14 +898,7 @@ trait LogicalTrees {
     }
 
     private def removeBounds(paramss: List[List[g.ValDef]]): List[List[g.ValDef]] = {
-      if (paramss.isEmpty) return Nil
-      val init :+ last = paramss
-      val hasImplicits = last.exists(_.mods.hasFlag(IMPLICIT))
-      val explicitss = if (hasImplicits) init else init :+ last
-      val implicits = if (hasImplicits) last else Nil
-      val limplicits = implicits.filter(_.name.startsWith(nme.EVIDENCE_PARAM_PREFIX))
```
Comment (xeno-by) [inline plugin/src/main/scala/org/scalameta/paradise/reflect/LogicalTrees.scala]: This line is an oversight. I should've inverted the condition in this filter. Could you submit this fix as a separate PR please?

**Comment by iTakeshi** [comment]
Ah, understood
